File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] a, int target) {
        int s = 0 ; 
        int e = a.length-1 ; 
        int ans = -1 ; 
        while(e>=s){
            int mid = s + ((e-s)>>1) ; 
            if(a[mid]==target){
                ans = mid ; 
                e = mid-1 ;
            }else 
            if(a[mid]>target){
                e  = mid-1 ;
            }else s = mid+1 ;
        }
        s = 0 ; 
        e = a.length -1 ; 
        int ans2 = -1 ; 
        while(e>=s){
            int mid = s + ((e-s)>>1) ; 
            if(a[mid]==target){
                ans2 = mid ; 
                s = mid+1 ;
            }else 
            if(a[mid]>target){
                e  = mid-1 ;
            }else s = mid+1 ;
        }
        return new int[]{ans,ans2} ;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

   int get(int nums[] ,int n ,  int target , int start){
    int s =  0  ; 
    int e = n-1 ;
    int ans  = -1 ;
    while(e>=s){
        int mid = s +(e-s)/2 ; 
        if(nums[mid]>target){
            e = mid- 1 ;
        }else if (nums[mid]<target){
            s = mid + 1 ;
        }
        else if(nums[mid]==target){
            ans = mid ; 
            if(start){
                e = mid - 1; 
            }else{
                s = mid + 1 ;
            }

        }
    }
    return ans ; 
   }
int* searchRange(int* nums, int numsSize, int target, int* returnSize) {
    int * a = (int*)malloc(sizeof(int)*2)  ; 
    a[0] = get(nums,numsSize,target,1) ; 
    a[1] = get(nums,numsSize,target,0) ;
    *returnSize = 2 ;
    return a ;
}